test/common/router/rds_impl_test.cc

```diff
@@ -268,6 +268,43 @@ TEST_F(RdsImplTest, FailureInvalidConfig) {
       "Unexpected RDS configuration (expecting foo_route_config): INVALID_NAME_FOR_route_config");
 }
 
+// rds and vhds configurations change together
+TEST_F(RdsImplTest, VHDSandRDSupdateTogether) {
+  setup();
+
+  const std::string response1_json = R"EOF(
+{
+  "version_info": "1",
+  "resources": [
+    {
+      "@type": "type.googleapis.com/envoy.config.route.v3.RouteConfiguration",
+      "name": "foo_route_config",
+      "vhds": {
```

Comment:
This test has only VHDS, not RDS + VHDS. Can you expand to cover the original bug?

Reply:
I think this test is fine, and i should explain the bug in other way

Reply:
The bug is triggered when both rds and vhds configurations change together. The test verifies the path when only vhds is present, and skips any checks about rds state.